src/catchup/ResolvedCatchupConfiguration.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include <cstdint>
+
+namespace stellar
+{
+
+class CatchupConfiguration;
+
+// Version of CatchupConfiguration class where value of toLedger is never
+// CatchupConfiguration::CURRENT - it is always real ledger numer.
+class ResolvedCatchupConfiguration
```

Comment:
This seems overkill, why not pass around a `CatchupConfiguration` and throw if it has a value that we don't expect. More code is not better.

Reply:
Possibly. I think that this code makes it compile-time error to pass wrong value (CatchupConfiguration::CURRENT) to a code that does not expects it. I can remove that if you wish and just add asserts to all code that currently receives ResolvedCatchupConfiguration.

Reply:
yeah it's common practice to have argument validation and throw a `bad_argument` exception